libs/api_common/store_helper_internal.cpp

```diff
@@ -161,6 +161,9 @@ _load_program_data_information(
         }
 
         auto program_type_name_string = ebpf_down_cast_from_wstring(std::wstring(program_type_name));
+        ebpf_free(program_type_name);
```

Comment:
To me it would seem to be simpler to just move the free at 272-273 outside the check for success.

Reply:
I am not happy with the current programming model in use in this code. It's too easy to accidentally leak memory. I think we should switch it to a [RAII](https://en.cppreference.com/w/cpp/language/raii) pattern.

Reply:
For C++ files like this one, I completely agree.

Reply:
For now I will just apply the suggested changes.